### ISSUE TITLE
Slf4jLogger name broken in 0.5.0

### ISSUE THREAD

**Issue by joroKr21** [zio/zio-logging#149]
Looks like somehow the changes in encoding broke the magic to set the logger name to the class name.
Now the logger name is always `zio.logging.LogAppender$$anon$2`
At a first glance I cannot tell why that changed.

**Comment by pshemass** [comment]
could you share code that you initialize `Layer`? it will help me reproduce this issue.

**Comment by pshemass** [comment]
I was able to reproduce this. sorry for this break I will try to fix this tomorrow

**Comment by joroKr21** [comment]
The setup is a bit complex but I'm not sure it matters:
```scala
val slf4j: ULayer[Logging] =
    Slf4jLogger.make { (context, message) =>
      ...
    }

val instrumented: RLayer[Has[Metrics] with Logging, Logging] =
    Logging.modifyLoggerM { logger =>
      for {
        metrics <- ZIO.service[Metrics]
        errors <- metrics.counter(...)
        warnings <- metrics.counter(...)
        logger <- logger.deriveM { originalContext =>
          for {
            localContext <- logger.logContext
            _ <- ZIO.whenCase(localContext.get(LogAnnotation.Level)) {
              case LogLevel.Error => errors.inc.orDie
              case LogLevel.Warn => warnings.inc.orDie
            }
          } yield originalContext
        }
      } yield logger
    }

val default: RLayer[Has[Metrics], Logging] =
    (ZLayer.service[Metrics] ++ slf4j) >>> instrumented
```

**Comment by joroKr21** [comment]
I think the problem is in `Slf4jLogger.make` and `classNameForLambda` that is trying to get the class of the lazy message.
Somehow it stopped working. The workaround is to provide a logger name explicitly (which might be a good idea anyway).

**Comment by pshemass** [comment]
no the problem is in LogAppender. since I separated formatiing and writing 
```
override def write(ctx: LogContext, msg: => A): UIO[Unit] =
            write0(ctx, format0.format(ctx, msg)).provide(env)
```
at this moment we lost source of lambda

**Comment by joroKr21** [comment]
Oh yeah I see

**Comment by pshemass** [comment]
fixed in 0.5.1
